lxd/storage_zfs.go

```diff
@@ -188,10 +188,12 @@ func (s *storageZfs) StoragePoolVolumeCreate() error {
 	logger.Infof("Creating ZFS storage volume \"%s\" on storage pool \"%s\".", s.volume.Name, s.pool.Name)
 
 	fs := fmt.Sprintf("custom/%s", s.volume.Name)
+	dataset := fmt.Sprintf("%s/custom/%s", s.pool.Name, s.volume.Name)
```

Comment:
So `s.pool.Name` is not correct as it only holds the name of the storage pool in the db not the on-disk name. You need to use `s.getOnDiskPoolName()`.